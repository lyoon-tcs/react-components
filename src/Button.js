/*
 * Copyright (c) 2006-present TrueChoice IP Holding Company, Inc.
 * All rights reserved.
 *
 * This software is the confidential and proprietary information of TrueChoice IP Holding Company, Inc.
 * ("Confidential Information").  You shall not disclose such Confidential Information and shall
 * use it only in accordance with the terms of the license agreement you entered into with the company.
 */

// dependencies
// local files
// import PropTypes from 'prop-types';

/**
 * @description - returns Button component
 * @param {string} props.buttonText - the button's text
 * @return {jsx} - the Button component to render
 */
const Button = (props) => {
  return <button>{props.buttonText}</button>;
};

// Button.propTypes = {
//   buttonText: PropTypes.string
// };

export default Button;
