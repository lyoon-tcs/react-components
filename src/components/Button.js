/*
 * Copyright (c) 2006-present TrueChoice IP Holding Company, Inc.
 * All rights reserved.
 *
 * This software is the confidential and proprietary information of TrueChoice IP Holding Company, Inc.
 * ("Confidential Information").  You shall not disclose such Confidential Information and shall
 * use it only in accordance with the terms of the license agreement you entered into with the company.
 */

// dependencies
import React from 'react';
import PropTypes from 'prop-types';
// local files
import styles from './Button.module.css';

/**
 * @description - returns Button component
 * @param {string} props.buttonText - the button's text
 * @param {string} props.backgroundColor - the button's background color
 * @param {string} props.color - the button's font color
 * @param {string} props.size - the button's size (extra padding & font-size)
 * @return {jsx} - the Button component to render
 */
const Button = ({
  buttonText = 'btn',
  backgroundColor = 'green',
  color = '#f2ff40',
  size = 'sm',
  handleClick
}) => {
  return (
    <button
      className={`${styles.button} ${styles[size]}`}
      style={{ backgroundColor, color }}
      onClick={handleClick}
    >
      {buttonText}
    </button>
  );
};

Button.propTypes = {
  buttonText: PropTypes.string,
  backgroundColor: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.oneOf(['sm', 'md', 'lg']),
  handleClick: PropTypes.func
};

export default Button;
