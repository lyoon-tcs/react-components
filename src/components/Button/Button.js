// dependencies
import React from 'react';
// import PropTypes from 'prop-types';

const Button = (props) => {
  return <button>{props.buttonText}</button>;
};

// Button.propTypes = {
//   buttonText: PropTypes.string
// };

export default Button;
