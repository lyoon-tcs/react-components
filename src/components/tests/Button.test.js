/*
 * Copyright (c) 2006-present TrueChoice IP Holding Company, Inc.
 * All rights reserved.
 *
 * This software is the confidential and proprietary information of TrueChoice IP Holding Company, Inc.
 * ("Confidential Information").  You shall not disclose such Confidential Information and shall
 * use it only in accordance with the terms of the license agreement you entered into with the company.
 */

// dependencies
import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';
// local files
import Button from '../Button';

afterEach(() => {
  cleanup();
});

describe('Button Component renders correctly', () => {
  const buttonText = 'hello world';
  render(<Button buttonText={buttonText} />);
  const buttonElement = screen.getByText(buttonText);

  test('should render button with proper text', () => {
    expect(buttonElement).toBeInTheDocument();
  });
  test('matches snapshot', () => {
    const tree = renderer.create(<Button buttonText={buttonText} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
