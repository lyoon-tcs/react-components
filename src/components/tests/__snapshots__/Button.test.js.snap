// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly matches snapshot 1`] = `
<button>
  hello world
</button>
`;
